plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseHiveConnectorTest.java

```diff
@@ -2102,6 +2104,122 @@ private void testBucketedTable(Session session, HiveStorageFormat storageFormat,
         assertFalse(getQueryRunner().tableExists(session, tableName));
     }
 
+    @Test(dataProvider = "bucketedFilterTestSetupsForSupportedDataTypes")
+    public void testFilterOnBucketedTable(BucketedFilterTestSetup testSetup)
+    {
+        String tableName = "test_filter_on_bucketed_table_" + randomTableSuffix();
+        assertUpdate(
+                """
+                CREATE TABLE %s (bucket_key %s, other_data double)
+                WITH (
+                    bucketed_by = ARRAY[ 'bucket_key' ],
+                    bucket_count = 5)
+                """.formatted(tableName, testSetup.trinoTypeName));
+
+        String values = testSetup.values.stream()
+                .map(value -> "(" + value + ", rand())")
+                .collect(joining(", "));
+        assertUpdate("INSERT INTO " + tableName + " VALUES " + values, testSetup.values().size());
+
+        // It will only read data from a single bucket instead of all buckets,
+        // so physicalInputPositions should be less than number of rows inserted (.
+        assertQueryStats(
+                getSession(),
+                """
+                SELECT count(*)
+                FROM %s
+                WHERE bucket_key = %s
+                """.formatted(tableName, testSetup.filterValue),
+                queryStats -> assertThat(queryStats.getPhysicalInputPositions()).isEqualTo(testSetup.expectedPhysicalInputRows),
```

Comment:
Please use getters on `testSetup`

Reply:
Why can't we use records builtin method like `testSetup.filterValue()`?

Reply:
Using record builtin is fine, is `testSetup.expectedPhysicalInputRows` a built-in ?

Reply:
no it would be `testSetup.expectedPhysicalInputRows()`

Reply:
But in this case, i'm getting error from error-prone

Reply:
Still, I'm getting this error. Do you know why?

```
Error:  /home/runner/work/trino/trino/plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseHiveConnectorTest.java:[8522,13] [UnusedVariable] The parameter 'expectedPhysicalInputRows' is never read.
```

Reply:
I've removed record